master/master.cfg

```diff
@@ -165,7 +165,86 @@ from buildbot.process.properties import Property
 from buildbot.process.properties import renderer
 from buildbot.process.properties import Interpolate
 
-from buildbot.process.properties import Interpolate
+
+class BuilderType:
+    """A class to encapsulate the settings for a specific Builder.
+       (Do not confuse with CMake's 'BUILD_TYPE', which is something else.)
+
+       It includes:
+       - Halide 'target' in the form of arch-bits-os
+       - LLVM branch to be used
+       - CMake vs Make
+
+       It doesn't currently include any 'features' because we don't currently
+       bake any in at build time.
+
+       It doesn't currently include the C++ compiler used (eg gcc7 vs gcc8 vs clang),
+       mainly because we currently never test with multiple compilers for a given
+       setup. (If we ever need to do so, compiler should be added to this.)
+    """
+
+    def __init__(self, arch, bits, os, llvm_branch, cmake=True, testbranch=False, distro=False):
+        assert arch in ['arm', 'x86']
+        assert bits in [32, 64]
+        assert os in ['linux', 'win', 'osx']
+        assert llvm_branch in [LLVM_TRUNK_BRANCH, LLVM_RELEASE_BRANCH, LLVM_OLD_BRANCH]
+        assert not (testbranch and distro)
+
+        self.arch = arch
+        self.bits = bits
+        self.os = os
+        self.llvm_branch = llvm_branch
+        self.cmake = cmake
+        self.testbranch = testbranch
+        # TODO: distro is an anachronism that should go away once proper CMake packaging lands
+        self.distro = distro
+
+    # The armbots aren't configured with Python at all,
+    # and supporting 32-bit Python on our 64-bit buildbots is painful
+    # (and usage of the python bindings on 32-bit hosts is unlikely anyway)
+    def handles_python():
+        return self.arch != 'arm' and self.bits != 32
+
+    def handles_hexagon():
+        return (builder_type.arch == 'x86'
+                and builder_type.bits == 64
+                and builder_type.os == 'linux'
+                and builder_type.llvm_branch == LLVM_TRUNK_BRANCH)
+
+    def handles_wasm():
+        return (builder_type.arch == 'x86'
+                and builder_type.bits == 64
+                and builder_type.os == 'linux'
+                and builder_type.llvm_branch == LLVM_TRUNK_BRANCH)
+
+    def has_nvidia():
```

Comment:
The windows buildbot also tests cuda (but I believe never calls this method, so it still would), but still, this is confusing.

Reply:
Fixed.